File: API/ios-tms.API/ViewModels/Quotes/QuoteIndexItemViewModel.cs
using Humanizer;
using iOS_TMS.Core.Domain;

namespace iOS_TMS.API.ViewModels.Quotes;

public class QuoteIndexItemViewModel
{
    public int Id { get; set; }
    public string Origin { get; set; }
    public string Destination { get; set; }
    public int Items { get; set; }
    public decimal Weight { get; set; }
    public decimal LowestRate { get; set; }
    public string LowestRateFormatted => LowestRate.ToString("0.00");
    public decimal HighestRate { get; set; }
    public string HighestRateFormatted => HighestRate.ToString("0.00");
    public int RateCount { get; set; }

    public static QuoteIndexItemViewModel From(Quote quote)
    {
        return new QuoteIndexItemViewModel
        {
            Id  = quote.Id,
            Origin = quote.Origin,
            Destination = quote.Destination,
            Items = quote.Items,
            Weight = quote.Weight,
            LowestRate = quote.QuoteRates.Select(r => r.Rate).Min(),
            HighestRate = quote.QuoteRates.Select(r => r.Rate).Max(),
            RateCount = quote.QuoteRates.Count
        };
    }
}

File: App/ios-tms/ViewModels/Quotes/QuoteIndexItemViewModel.swift
//
//  QuoteIndexItemViewModel.swift
//  ios-tms
//
//  Created by Kyle Morton on 9/17/22.
//

import Foundation

struct QuoteIndexItemViewModel : Identifiable, Codable {

    var id: Int
    var origin: String
    var destination: String
    var items: Int
    var weight: Decimal
    var weightFormatted: String
    var lowestRate: Decimal
    var lowestRateFormatted: String
    var highestRate: Decimal
    var highestRateFormatted: String
    var rateCount: Int
    var pickupDateFormatted: String
    
    init(id: Int, origin: String, destination: String, items: Int, weight: Decimal, lowestRate: Decimal, lowestRateFormatted: String, highestRate: Decimal, highestRateFormatted: String, pickupDateFormatted: String, rateCount: Int) {
        self.id = id
        self.origin = origin
        self.destination = destination
        self.items = items
        self.weight = weight
        self.lowestRate = lowestRate
        self.lowestRateFormatted = lowestRateFormatted
        self.highestRate = highestRate
        self.highestRateFormatted = highestRateFormatted
        self.rateCount = rateCount
        self.weightFormatted = "\(weight)"
        self.pickupDateFormatted = pickupDateFormatted
    }
    
}
